snuba/snuba_migrations/functions/0004_functions_v2.py

```diff
@@ -0,0 +1,232 @@
+                table_name=table_name,
+                column=new_column.column,
+                after=new_column.after,
+                target=target,
+            )
+            for table_name, target in [
+                (self.local_materialized_table, OperationTarget.LOCAL),
+                (self.dist_materialized_table, OperationTarget.DISTRIBUTED),
+            ]
+            for new_column in new_columns
+        ] + [
+            operations.CreateMaterializedView(
+                storage_set=self.storage_set,
+                view_name=self.local_view_table,
+                destination_table_name=self.local_materialized_table,
+                columns=all_columns,
+                query=self.__MATVIEW_STATEMENT,
+                target=OperationTarget.LOCAL,
+            ),
+        ]
+
+    def backwards_ops(self) -> Sequence[operations.SqlOperation]:
+        return [
+            operations.DropTable(
+                storage_set=self.storage_set,
+                table_name=self.local_view_table,
+                target=operations.OperationTarget.LOCAL,
+            )
+        ] + [
+            operations.DropColumn(
+                storage_set=self.storage_set,
+                table_name=table_name,
+                column_name=new_column.column.name,
+                target=target,
+            )
+            for table_name, target in [
+                (self.dist_materialized_table, OperationTarget.DISTRIBUTED),
+                (self.local_materialized_table, OperationTarget.LOCAL),
+            ]
+            for new_column in new_columns
+        ]
+
+    @property
+    def __MATVIEW_STATEMENT(self) -> str:
+        return f"""
+            SELECT
+                project_id,
+                transaction_name,
+                name,
+                package,
+                path,
+                fingerprint,
+                parent_fingerprint,
+                depth,
+                is_application,
+                profiling_type,
+                platform,
+                environment,
+                release,
+                os_name,
+                os_version,
+                toDateTime({self.data_granularity} * intDiv(toUnixTimestamp(timestamp), {self.data_granularity})) AS timestamp,
+                retention_days,
+                countState(arrayJoin(durations) AS duration) AS count,
+                quantilesState(0.5, 0.75, 0.9, 0.95, 0.99)(duration) AS percentiles,
+                minState(duration) AS min,
+                maxState(duration) AS max,
+                avgState(duration) AS avg,
+                sumState(duration) AS sum,
+                argMaxState(profile_id, duration) as worst,
+                argMaxState((profile_id, thread_id, start_timestamp, end_timestamp), duration) as worst_v2,
+                groupUniqArrayState(5)(profile_id) as examples,
+                groupUniqArrayState(5)((profile_id, thread_id, start_timestamp, end_timestamp)) as examples_v2
```

Comment:
Hm... I just realized that in order to truly drop the existing columns `worst` and `examples`, we'll need yet another migration where we create another materialized view but without those 2 aggregations.

Reply:
Yes, we can think about that in ~90 days.

For now it's handy to have both since depending on the time-range we might still need to access examples for which we only have data in the older `worst` and `examples` columns.